cyclonedx/model/component.py

```diff
@@ -300,20 +534,24 @@ def to_package_url(self) -> PackageURL:
         Returns:
             `packageurl.PackageURL` instance which represents this Component.
         """""
-        return PackageURL(
-            type=self._package_url_type,
-            namespace=self._namespace,
-            name=self._name,
-            version=self._version,
-            qualifiers=self._qualifiers,
-            subpath=self._subpath
-        )
+        return PackageURL(type=self.__purl_type, namespace=self.group, name=self.name, version=self.version,
+                          qualifiers=self.__purl_qualifiers, subpath=self.__purl_subpath)
 
     def __eq__(self, other: object) -> bool:
         if isinstance(other, Component):
-            return other.get_purl() == self.get_purl()
-        else:
-            raise NotImplementedError
+            return other.purl == self.purl
```

Comment:
equal PURL does not make two components equal.
two components with the same name could be equal, but dont necessarily must be.
purl seams to be just an arbitrary property of a component.

Reply:
Absolutely correct.

Have now implemented `__hash__` and `__eq__` uses `hash()`.